plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal.php

```diff
@@ -340,6 +340,20 @@ public function maybe_onboard_with_transact() {
 		$transact_account_manager->do_onboarding();
 	}
 
+	/**
+	 * Check if the gateway is available for use.
+	 *
+	 * @return bool
+	 */
+	public function is_available() {
+		// For Orders v2, require a valid email address to be set up in the gateway settings.
+		if ( $this->should_use_orders_v2() && ! is_email( $this->email ) ) {
```

Comment:
Is it worth adding a quick empty check as well as a full `is_email()` check? We know an empty value can't be a valid email.
```suggestion
		if ( $this->should_use_orders_v2() && ( empty( $this->email ) || ! is_email( $this->email ) ) ) {
```

Reply:
Sounds good. Accepting the suggested change 👌